wamr-compiler/main.c

```diff
@@ -516,7 +539,13 @@ main(int argc, char *argv[])
             PRINT_HELP_AND_EXIT();
     }
 
-    if (argc == 0 || !out_file_name)
+    /* Normal usage, generate output file */
+    if (!use_dummy_wasm && (argc == 0 || !out_file_name))
+        PRINT_HELP_AND_EXIT();
+
+    /* When use wamrc to print architecture related help info, use dummy_wasm
+     * and don't generate anything */
+    if (use_dummy_wasm && out_file_name)
```

Comment:
My understanding is that the below commands are allowed:
```bash
wamrc --target=help
wamrc --target=help -o test.aot test.wasm
wamrc --target=help -o test.aot
wamrc --target=help test.wasm
```
Should remove these lines?

Reply:
My modification will disable passing input/output filename while printing help info:
```shell
# can't do this, will print usage 
wamrc --target=help -o test.aot test.wasm
wamrc --target=help -o test.aot
wamrc --target=help test.wasm
```
Since only want help info, there is no point in passing an input/output file name.